Software/lib/GNSE_BSP/GNSE_bsp_gpio.h

```diff
@@ -132,6 +132,16 @@ typedef enum
 #define LOAD_SWITCH2_GPIO_CLK_ENABLE() __HAL_RCC_GPIOC_CLK_ENABLE()
 #define LOAD_SWITCH2_GPIO_CLK_DISABLE() __HAL_RCC_GPIOC_CLK_DISABLE()
 
+/*!
+ * Load Switch delay time defines
+ * Note that all include the settling time of the Load Switch
+ */
+#define LOAD_SWITCH_ON_TIME_MS 1U // This value is rounded up from 120-200 us
```

Comment:
It's great that you did this exercise
Now there is an issue with the implementation

1. You should define a max delay time of all the sensors connected to this load switch as you don't know if the user will init the accelerometer or the secure element something Logically it would be 
2. You didn't account for environment temperature variations which risks functionality
3. There are two load switches and this confuses the user, as you consider the flash it's own load switch which is not ok

This can break functionality easily for optimization sake and that is not ok, especially for a public repo.

I think you should have two delays and comment your findings to explain the delay value.

```c
# define LOAD_SWITCH1_DELAY_MS x
# define LOAD_SWITCH2_DELAY_MS x
```